src/cli.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! CLI implementation.
+
+use crate::common::{PROJECT_AUTHOR, PROJECT_DESC, PROJECT_NAME, PROJECT_VERSION};
+use clap::{App, AppSettings, Arg};
+
+#[derive(Debug)]
+pub enum Action {
```

Comment:
In the future, `Action` should probably be replaced by [`NativeOperation`](https://docs.rs/parsec-interface/0.19.0/parsec_interface/operations/enum.NativeOperation.html) to not have to duplicate similar information. This module would create the correct instance of `NativeOperation` from the command entered by the user.
Having an automatic translation (as a procedural macro maybe?) of each of the `Operation` into command line arguments for `clap` would be phenomenal.

Reply:
> Having an automatic translation (as a procedural macro maybe?) of each of the `Operation` into command line arguments for `clap` would be phenomenal.

I don't think that's a good idea for some of the operations at least, especially the ones that take very rich data structures. Imagine having to write the whole `KeyAttributes` structure on the command line. It's probably more sensible to expect stuff like that to be written in a TOML file or something similar, and we parse it from there.

Hence, I'm not sure just automating the whole translation process makes that much sense, unless we can specify, for example, which parameters of a command should be taken from files, and how the file should be converted to the parameter in the first place. Even for "very simple" data like a file that you want to encrypt, it's probably more reasonable to let the user decide if they want to pipe the contents through the CLI or have an option to select the file based on its path (and potentially write a signature for it in a different file, instead of just blindly writing stuff to stdout).

Reply:
That's true, I did not think about `KeyAttributes`, that would be very painful to write all down. Maybe more generally the CLI should limit itself to the highest abstraction level of the client? When we have features for parameters discovery and simple abstraction over complicated operations, then the CLI can have commands over that.